aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,182 @@
+  // The parent resource shared by all books being retrieved.
+  // Format: publishers/{publisher}
+  // If this is set, the parent of all of the books specified in `paths`
+  // must match this field.
+  string parent = 1 [
+    (google.api.resource_reference) = {
+      child_type: "library.com/Book"
+    }];
+
+  // The paths of the books to retrieve.
+  // A maximum of 1000 books can be retrieved in a batch.
+  // Format: publishers/{publisher}/books/{book}
+  repeated string paths = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.com/Book"
+    }];
+}
+```
+
+- The request schema **must** match the method name, with `Request` suffix.
+- A `parent` field **should** be included, unless the resource being retrieved
+  is a top-level resource, to facilitate inclusion in the URI as well to permit
+  a single permissions check. If a caller sets this field, and the parent
+  collection in the path of any resource being retrieved does not match, the
+  request **must** fail.
+  - This field **should** be required if only 1 parent per request is allowed.
+  - The field **should** identify the [resource type][aep-122-parent] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- There **must not** be a body key in the `google.api.http` annotation.
+
+{% tab oas %}
+
+{% sample 'batchget.oas.yaml', 'paths' %}
+
+- The `paths` parameter **must** be a query parameter which accepts an array of
+  resource paths specifying the resources to retrieve. The parameter **should**
+  be named `paths`.
+  - If no resource paths are provided, the API **should** error with status
+    code `400 Bad Request`.
+  - If the collection in the path of any resource does not match the collection
+    of the request URL, the request **must** fail.
+- The method definition **must not** have a `requestBody` defined.
+
+{% endtabs %}
+
+### Response
+
+The response for a batch get method **should** be specified with the following
+pattern:
+
+- The response schema **must** match the method name, with `Response` suffix.
+- The response schema **must** have a single array property where each item is
+  either a retrieved resource or an error structure describing an error that
+  occurred attempting to retrieve the resource. The error alternative would
+  only be present for non-transactional batch gets.
+- The order of resources/error objects in the response **must** be the same as
+  the paths in the request.
+- The array of resources **must** be named `results` and contain resources with
+  no additional wrapping.
+- There must not be a `nextPageToken` field as batch get operations are not
+  pageable.
+
+{% tab proto -%}
+
+```proto
+message BatchGetBooksResponse {
+  // Books requested.
+  repeated Book books = 1;
+}
+```
+
+{% tab oas %}
+
+{% sample 'batchget.oas.yaml', 'paths./publishers/{publisherId}/books:BatchGet.get.responses.200.content.application/json' %}
```

Comment:
```suggestion
{% sample 'batchget.oas.yaml', 'paths./publishers/{publisherId}/books:BatchGet' %}
```
This is broken for me - I get a template syntax error. The site-generator matching is pretty rudimentary, I think it only does a substring match of some kind. Something to look into CC @rambleraptor, but the only blocker is the breaking of rendering.

![20241018_10h40m15s_grim](https://github.com/user-attachments/assets/5f574e78-541c-4b34-9184-f26c5349ce7e)